sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -388,8 +403,82 @@ public Date getLatestFollowUpEndDateByUuid(String uuid) {
 		cq.orderBy(cb.desc(contactRoot.get(Contact.FOLLOW_UP_UNTIL)));
 
 		List<PersonFollowUpEndDto> resultlist = em.createQuery(cq).getResultList().stream().distinct().collect(Collectors.toList());
-		return resultlist.get(0).getLatestFollowUpEndDate();
+		if (resultlist.isEmpty()) {
+			return null;
+		} else {
+			return resultlist.get(0).getLatestFollowUpEndDate();
+		}
+
+	}
+
+	@Override
+	public FollowUpStatus getMostRelevantFollowUpStatusByUuid(String uuid) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<FollowUpStatusDto> cq = cb.createQuery(FollowUpStatusDto.class);
+
+		Root<Contact> contactRoot = cq.from(Contact.class);
+		Join<Contact, Person> personContactJoin = contactRoot.join(Contact.PERSON, JoinType.LEFT);
+		Predicate contactFilter = contactService.createUserFilter(cb, cq, contactRoot);
+		if (uuid != null) {
+			contactFilter = PersonService.and(cb, contactFilter, cb.equal(personContactJoin.get(Person.UUID), uuid));
+		}
+		if (contactFilter != null) {
+			cq.where(contactFilter);
+		}
+		cq.multiselect(personContactJoin.get(Person.UUID), contactRoot.get(Contact.FOLLOW_UP_STATUS));
+		List<FollowUpStatusDto> contactResultList = em.createQuery(cq).getResultList().stream().distinct().collect(Collectors.toList());
+
+		cq = cb.createQuery(FollowUpStatusDto.class);
+		Root<Case> caseRoot = cq.from(Case.class);
+		Join<Case, Person> personCaseJoin = caseRoot.join(Case.PERSON, JoinType.LEFT);
+		Predicate caseFilter = caseService.createUserFilter(cb, cq, caseRoot);
+		if (uuid != null) {
+			caseFilter = PersonService.and(cb, caseFilter, cb.equal(personCaseJoin.get(Person.UUID), uuid));
+
+		}
+		if (caseFilter != null) {
+			cq.where(caseFilter);
+		}
+		cq.multiselect(personCaseJoin.get(Person.UUID), caseRoot.get(Case.FOLLOW_UP_STATUS));
+		List<FollowUpStatusDto> caseResultList = em.createQuery(cq).getResultList().stream().distinct().collect(Collectors.toList());
+
+		List<FollowUpStatusDto> resultList = Stream.concat(contactResultList.stream(), caseResultList.stream()).collect(Collectors.toList());
 
+		if (resultList.isEmpty()) {
+			return null;
+		} else {
+			for (FollowUpStatusDto status : resultList) {
+				if (FollowUpStatus.FOLLOW_UP.equals(status.getFollowUpStatus())) {
+					return FollowUpStatus.FOLLOW_UP;
+				}
+			}
+			if (listOnlyContainsStatus(resultList, FollowUpStatus.CANCELED)) {
+				return FollowUpStatus.CANCELED;
+			} else if (listOnlyContainsStatus(resultList, FollowUpStatus.COMPLETED)) {
+				return FollowUpStatus.COMPLETED;
+			} else if (listOnlyContainsStatus(resultList, FollowUpStatus.LOST)) {
+				return FollowUpStatus.LOST;
+			} else {
+				return FollowUpStatus.NO_FOLLOW_UP;
+			}
+		}
+	}
+
+	private boolean listOnlyContainsStatus(List<FollowUpStatusDto> list, FollowUpStatus prameterStatus) {
+		if (list.isEmpty()) {
+			return false;
+		}
+		if (prameterStatus == null) {
```

Comment:
I wouldn't throw `IllegalArgumentException` in a private function and also the exception message is too generic, it doesn't tell which parameter is wrong.

Reply:
I'm going to replace it with an assert statement.